### DISCUSSION TITLE
MongoDB soll nur einen Eintrag im Model haben

### DISCUSSION THREAD

**Discussion by FreytagDom** [neuefische/questions#228] [Web] [answered]
Hallo zusammen,

ich habe grade das Problem das ich nur einen Eintrag in MongoDB im Model haben möchte. Sobald man über das Eingabeformular neue Daten eingibt sollen die vorhandenen überschrieben werden. 
Hoffe einer kann mir helfen.
Danke schon mal. 

Hello all,

I just have the problem that I only want to have one entry in MongoDB in the model. As soon as I enter new data in the input form, the existing data is overwritten. 
I hope someone can help me.
Thanks already. 
![Bildschirmfoto 2022-10-07 um 12 05 17](https://user-images.githubusercontent.com/109357762/194529237-731b5825-040a-43fb-8dd4-72c8d167d02f.png)

![Bildschirmfoto 2022-10-07 um 12 05 26](https://user-images.githubusercontent.com/109357762/194529263-8fa84304-232a-4979-b6d3-8ea8fa8ddcf2.png)

**Comment by ThomasKirst** [comment]
Hi Dominic,
can you please provide some example code / a link to a PR which shows how you are trying to achieve this?
–Thomas

EDIT:
You can provide your own ID to a MongoDB document. See the mongoose documentation here:
–> https://mongoosejs.com/docs/guide.html#_id

You could provide a simple number for your one document:

```js
const schema = new Schema({ _id: Number });
const Model = mongoose.model('Test', schema);

const doc = new Model();
await doc.save(); // Throws "document must have an _id before saving"

doc._id = 1;
await doc.save(); // works
```

You will need some kind of logic which checks if the document already exists (`findOneById(1)`).
If the document doesn´t exist yet you will need to create it first.
If it does exist you will have to perform an update (`findByIdAndUpdate(1, {...})`).
–> https://mongoosejs.com/docs/api.html#model_Model-findByIdAndUpdate

There is also a convenient function called [findOneAndUpdate()](https://mongoosejs.com/docs/api.html#model_Model-findOneAndUpdate).
This function takes an option (`{upsert: true}`) which CREATES the document if it doesn´t exist yet.

`[options.upsert=false] «Boolean» if true, and no documents found, insert a new document`

**Reply by FreytagDom** [reply]
https://github.com/FreytagDom/capstone-project/pull/12

**Reply by ThomasKirst** [reply]
I suggested two code changes in your PR that should solve your problem:
–> https://github.com/FreytagDom/capstone-project/pull/12#pullrequestreview-1134484122

**Comment by FreytagDom** [comment, marked as answer]
Problem gelöst.

create durch updateMany ersetzt 

<img width="874" alt="Bildschirmfoto 2022-10-07 um 15 57 33" src="https://user-images.githubusercontent.com/109357762/194570938-ad6bdebc-dba7-4396-bfd2-1c5ca1858dc9.png">
